charts/argo-rollouts/values.yaml

```diff
@@ -210,14 +210,14 @@ controller:
   # -- Configures 3rd party metric providers for controller
   ## Ref: https://argo-rollouts.readthedocs.io/en/stable/analysis/plugins/
   metricProviderPlugins: {}
-    # metricProviderPlugins: |-
+    # metricProviderPlugins:
     #   - name: "argoproj-labs/sample-prometheus" # name of the plugin, it must match the name required by the plugin so that it can find its configuration
     #     location: "file://./my-custom-plugin" # supports http(s):// urls and file://
```

Comment:
Thank you for your PR.

Originally we needed to define like this, so `object` was ok.
```yaml
metricProviderPlugins:
  metricProviderPlugins: |-
    - name: "argoproj-labs/sample-prometheus" # name of the plugin, it must match the name required by the plugin so that it can find its configuration
      location: "file://./my-custom-plugin" # supports http(s):// urls and file://
```
But now we define like this.
```yaml
metricProviderPlugins:
  - name: "argoproj-labs/sample-prometheus" # name of the plugin, it must match the name required by the plugin so that it can find its configuration
    location: "file://./my-custom-plugin" # supports http(s):// urls and file://
```
So I wonder it's better to change type as `list` ?
```diff
-  metricProviderPlugins: {}
+  metricProviderPlugins: []
-    # metricProviderPlugins: |-
-    # metricProviderPlugins:
-    # - name: "argoproj-labs/sample-prometheus" # name of the plugin, it must match the name required by the plugin so that it can find its configuration
-    #   location: "file://./my-custom-plugin" # supports http(s):// urls and file://
+    # - name: "argoproj-labs/sample-prometheus" # name of the plugin, it must match the name required by the plugin so that it can find its configuration
+    #   location: "file://./my-custom-plugin" # supports http(s):// urls and file://
```

```

```

Reply:
I had always assumed the whole indented block was to be uncommented and de-dented.. :)  
ala
```
values:
    # metricProviderPlugins: |-
    #   - name: "argoproj-labs/sample-prometheus" # name of the plugin, it must match the name required by the plugin so that it can find its configuration
    #     location: "file://./my-custom-plugin" # supports http(s):// urls and file://
```
becomes
```
values:
   metricProviderPlugins: |-
     - name: "argoproj-labs/sample-prometheus" # name of the plugin, it must match the name required by the plugin so that it can find its configuration
       location: "file://./my-custom-plugin" # supports http(s):// urls and file://
```

mostly since the "default" uncommented version was something you replace w/ the commented out things.

This looks fine I'll make the change on Monday

Reply:
>I had always assumed the whole indented block was to be uncommented and de-dented.. :)

I am the one who added this attribute in argo-helm as following other existing ones. https://github.com/argoproj/argo-helm/blob/main/charts/argo-rollouts/values.yaml#L30-L39
*Though I should've implemented a better way, trim redundant `metricProviderPlugins` in configmap.yaml.

```yaml
metricProviderPlugins: 
    metricProviderPlugins: |-
      - name: "argoproj-labs/sample-prometheus" # name of the plugin, it must match the name required by the plugin so that it can find its configuration
        location: "file://./my-custom-plugin" # supports http(s):// urls and file://
```